### PR TITLE
Implement a ViewModel for Reviews

### PR DESCRIPTION
**Pull request by LouisJamesVasseur**
### The Content of the PR

This PR has the objective of creating a ViewModel capable of representing View objects. Please note the following: this is the first of a series of 3 PRs to implement Reviews, which will be followed by a UI-based PR to implement ReviewScreen for a User to add a review on a parking spot and another PR for implementing the ViewModel in other parts of the app. Please note therefore that this ViewModel is very much subject to change.

### Why ?
Since a main component of the app is for users to find good parking spots, this is an essential feature to add and will be expanded upon later.

### How ?
The provided ViewModel works in a fundamentally similar way to that of Parking: a FirestoreDatabase (though with different attributes for the data class, of course) with similar provided functions, though some more are added.

### Testing

A few Android tests are provided through which you can test functionality of the Firestore and ViewModel. Some example reviews are provided as well

### Code Coverage 
Screenshot 2024-10-16 at 13 37 12

### CODE REVIEW

**Review by Armetiis — COMMENTED** [review]
Good job! I understand that the get methods for the viewModel are the subject of a future PR. I left a comment on the minor issues that I spotted. You have my approval, but please fix those before proceding to merge.

**Review by Armetiis — APPROVED** [review]

File: app/src/main/java/com/github/se/cyrcle/model/review/Review.kt
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/review/Review.kt:7]: I think this will make more sense when I show the screen too. The idea was for the screen to display something similar to stars, and I've personally seen a good amount of apps allow half-stars. I'll leave it as a double for now and we can decide whether or not we make that change then

**Review by Pijukai — APPROVED** [review]
Nothing to add to the minor issues that @Armetiis brought up. Good Job !

**Pull request merged**
